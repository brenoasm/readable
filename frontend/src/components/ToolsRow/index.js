import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Button from '../Button/';

const propTypes = {
  options: PropTypes.arrayOf(PropTypes.object),
  selectedOption: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onFilterChange: PropTypes.func,
  showModal: PropTypes.func,
};

const defaultProps = {
  options: [],
  selectedOption: '',
  onFilterChange: () => {},
  showModal: () => {}
};

const StyledToolsRow = styled.div`
  display: flex;
  justify-content: flex-end;

  button {
    border-radius: 3px;
    margin-right: 10px;

    a {
      text-decoration: none;
    }
  }

  & > span {
    align-self: center;

    select {
      height: 34px;
      cursor: pointer;
    }
  }
`;

const ToolsRow = ({ options, selectedOption, onFilterChange, showModal }) => (
  <StyledToolsRow>
    <Button text="Adicionar Postagem" handleClick={() => showModal()} />
    <span>
      <label htmlFor="select-sort-method">Filtrar por: </label>
      <select
        value={selectedOption}
        name="select-sort-method"
        onChange={event => onFilterChange(event.target.value)}
      >
        {Array.isArray(options) &&
          options.map(op => (
            <option key={op.value} value={op.value}>
              {op.text}
            </option>
          ))}
      </select>
    </span>
  </StyledToolsRow>
);

ToolsRow.propTypes = propTypes;
ToolsRow.defaultProps = defaultProps;

export default ToolsRow;
