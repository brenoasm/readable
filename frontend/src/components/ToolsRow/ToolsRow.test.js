/* eslint-disable */

import React from 'react';

import ToolsRow from './';

describe('ToolsRow', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ToolsRow />);
  });

  afterEach(() => {
    wrapper.unmount;
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a NewPostButton component', () => {
    expect(wrapper.find('NewPostButton')).toHaveLength(1);
  });
});
