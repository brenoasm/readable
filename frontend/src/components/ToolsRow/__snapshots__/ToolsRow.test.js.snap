// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolsRow matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToolsRow
    onFilterChange={[Function]}
    options={Array []}
    selectedOption=""
    showModal={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <NewPostButton
          handleClick={[Function]}
/>,
        <span>
          <label
                    htmlFor="select-sort-method"
          >
                    Filtrar por: 
          </label>
          <select
                    name="select-sort-method"
                    onChange={[Function]}
                    value=""
          >
                    
          </select>
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="select-sort-method"
>
              Filtrar por: 
</label>,
            <select
              name="select-sort-method"
              onChange={[Function]}
              value=""
>
              
</select>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Filtrar por: ",
              "htmlFor": "select-sort-method",
            },
            "ref": null,
            "rendered": "Filtrar por: ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "name": "select-sort-method",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": Array [],
            "type": "select",
          },
        ],
        "type": "span",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <NewPostButton
            handleClick={[Function]}
/>,
          <span>
            <label
                        htmlFor="select-sort-method"
            >
                        Filtrar por: 
            </label>
            <select
                        name="select-sort-method"
                        onChange={[Function]}
                        value=""
            >
                        
            </select>
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="select-sort-method"
>
                Filtrar por: 
</label>,
              <select
                name="select-sort-method"
                onChange={[Function]}
                value=""
>
                
</select>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Filtrar por: ",
                "htmlFor": "select-sort-method",
              },
              "ref": null,
              "rendered": "Filtrar por: ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "name": "select-sort-method",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": Array [],
              "type": "select",
            },
          ],
          "type": "span",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
